tensorflow/core/kernels/mkl_aggregate_ops.cc

```diff
@@ -38,202 +38,192 @@ class MklAddNOp : public OpKernel {
-      // Memory descriptor for 1st input
-      srcs_pd.push_back(memory::primitive_desc(common_md, cpu_engine));
-      // Memory descriptor for 2nd input
-      srcs_pd.push_back(memory::primitive_desc(common_md, cpu_engine));
-      auto sum_pd = sum::primitive_desc(dst.GetUsrMemDesc(), coeff, srcs_pd);
-
-      // Now we setup resources for primitive execution.
-      // First, we need to check if any of the inputs need to be reordered as
-      // per the logic described above. Since output will be in MKL format if
-      // atleast one input is in MKL format, we choose output descriptor for
-      // reorder.
       std::vector<primitive::at> inputs;
-      // Check if actual input format of the tensor is different than common_pd
-      // we told MKLDNN. In that case, we will need reorder.
-      src1.CheckReorderToOpMem(srcs_pd[0]);
-      src2.CheckReorderToOpMem(srcs_pd[1]);
-      inputs.push_back(src1.GetOpMem());
-      inputs.push_back(src2.GetOpMem());
-
-      // Allocate output tensor now.
-      Tensor* dst_tensor = nullptr;
-      MklDnnShape output_mkl_shape;
-      TensorShape output_tf_shape;
 
-      if (input2_in_mkl_format || input1_in_mkl_format) {
-        output_mkl_shape.SetMklTensor(true);
-        auto output_pd = dst.GetUsrMemPrimDesc();
-        output_mkl_shape.SetMklLayout(&output_pd);
-        output_mkl_shape.SetElemType(MklDnnType<T>());
-        if (input1_in_mkl_format) {
-          output_mkl_shape.SetTfLayout(src1_dims_size,
-                                       src1_mkl_shape.GetSizesAsMklDnnDims(),
-                                       src1_mkl_shape.GetTfDataFormat());
+      MklDnnData<T> dst(&cpu_engine);
+
+      MklDnnData<T> src = MklDnnData<T>(&cpu_engine);
+      bool has_mkl_input = false;
+      int mkl_input_index = FindMKLInputIndex(ctx);
+      if (mkl_input_index >= 0) {
+        has_mkl_input = true;
+        GetMklShape(ctx, mkl_input_index, &mkl_shape);
+      }
+
+      // Create memory descriptor for MKL-DNN.
+      // If all input in Tensorflow format, create block memory descriptor,
+      // else convet TF format to MKL memory descriptor
+      for (int src_idx = 0; src_idx < num_inputs; src_idx++) {
+        MklDnnShape src_mkl_shape;
+        GetMklShape(ctx, src_idx, &src_mkl_shape);
+        memory::desc md({}, memory::data_undef, memory::format_undef);
+        src = MklDnnData<T>(&cpu_engine);
+        const Tensor& src_tensor = MklGetInput(ctx, src_idx);
+
+        if (src_mkl_shape.IsMklTensor()) {
+          md = src_mkl_shape.GetMklLayout();
         } else {
-          output_mkl_shape.SetTfLayout(src2_dims_size,
-                                       src2_mkl_shape.GetSizesAsMklDnnDims(),
-                                       src2_mkl_shape.GetTfDataFormat());
+          if (has_mkl_input) {
+            // MKL input has the data format information.
+            memory::format mkl_data_format = mkl_shape.GetTfDataFormat();
+            auto tf_data_format =
```

Comment:
This and the line above can be done after `GetMklShape(ctx, mkl_input_index, &mkl_shape);` in line 167.